### PR TITLE
[bitnami/kafka] Clean Incorrect Warning Messages for 3.0/3.1/3.2/3.3

### LINKED ISSUE #6172: The WARN log seems incorrect
### Name and Version

bitnami/kafka:3.2.0

### What steps will reproduce the bug?

1. docker-compose look like this:

```yaml
kafka:
    network_mode: host
    image: bitnami/kafka:3.2.0
    environment:
      - KAFKA_ENABLE_KRAFT=yes
      - KAFKA_CFG_PROCESS_ROLES=broker,controller
      - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://localhost:9092
      - KAFKA_BROKER_ID=1
      - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=1@127.0.0.1:9093
      - ALLOW_PLAINTEXT_LISTENER=yes
      - KAFKA_KRAFT_CLUSTER_ID=V5t6x-LERXi-eUBdy_vacg
      - KAFKA_HEAP_OPTS=-Xmx512M -Xms512M

```

2. execute `docker compose up kafka`

### What is the expected behavior?

No Warning Log

### What do you see instead?

I receive these logs from the console

```
docker-kafka-1  | kafka 04:34:27.11 WARN  ==> KAFKA_CFG_BROKER_ID Must match what is set in KAFKA_CFG_CONTROLLER_QUORUM_VOTERS
docker-kafka-1  | kafka 04:34:27.11 WARN  ==> KAFKA_CFG_CONTROLLER_QUORUM_VOTERS must match brokers set with KAFKA_CFG_BROKER_ID
docker-kafka-1  | kafka 04:34:27.11 WARN  ==> KAFKA_CFG_PROCESS_ROLES must include 'controller' for KRaft
docker-kafka-1  | kafka 04:34:27.11 WARN  ==> KAFKA_CFG_LISTENERS must include a listener for CONTROLLER
docker-kafka-1  | kafka 04:34:27.11 WARN  ==> You set the environment variable ALLOW_PLAINTEXT_LISTENER=yes. For safety reasons, do not use this flag in a production environment.
```
I understand that the last warning is correct, but others warnings are not correct

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jiak94**
### Description of the change

Clean incorrect warning message

### Benefits

logs will not be confused

### Applicable issues

  - fixes #6172

### CODE REVIEW

**Comment by rafariossaa** [comment]
Hi,
  Sorry, I have just updated the contributing guidelines because there was a note missing regarding splitting a PR if the changes affect several repositories.

**Comment by jiak94** [comment]
> Hi, Sorry, I have just updated the contributing guidelines because there was a note missing regarding splitting a PR if the changes affect several containers. Do you mind splitting this PR ?

Please see #7522  and #7523

**Comment by rafariossaa** [comment]
Hi,
  Thanks. 
  Let's wait until they get reviewed and merged.

**Comment by dimaqq** [comment]
@rafariossaa how to resurrect this PR?

**Comment by carrodher** [comment]
Sorry for the delay, my teammate @rafariossaa  will review it during the next few days

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh
Comment (rafariossaa) [inline bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:243]: I would suggest to keep the old IFS and restoring it after.
Other part of the code could relay on the default value for IFS and this would interfere with that:
```
old_IFS=$IFS
IFS="something_new"
run_your_code
IFS=${old_IFS}
````

What do you think ?

**Review by rafariossaa — COMMENTED** [review]
Sorry for the delay, could you take a look to my comment ?

**Comment by carrodher** [comment]
As suggested in https://github.com/bitnami/containers/pull/7522 and https://github.com/bitnami/containers/pull/7523, you can add the same changes affecting all the branches into a single PR.

Since this one was already reviewed by @rafariossaa you can edit the title and use this one to collect the changes related to all the active Kafka branches.

Please note there is [a new 3.3 branch](https://github.com/bitnami/containers/tree/main/bitnami/kafka/3.3/debian-11).

**Comment by jiak94** [comment]
@carrodher can you help to check why VIB Verify for 3.2 and 3.3 failed?

**Comment by carrodher** [comment]
> @carrodher can you help to check why VIB Verify for 3.2 and 3.3 failed?

Thanks for applying the changes. Don't worry about the CI failures at this point, we are having some networking issues and the failure is totally unrelated to your changes

File: bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh
Comment (rafariossaa) [inline bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:264]: What about reordering it like this ?

```suggestion
if [[ -z "$KAFKA_CFG_BROKER_ID" ]]; then
    print_validation_error "KRaft requires KAFKA_CFG_BROKER_ID to be set for the quorum controller"
fi
if [[ -z "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS" ]]; then
    print_validation_error "KRaft requires KAFKA_CFG_CONTROLLER_QUORUM_VOTERS to be set"
fi

if [[ -n "$KAFKA_CFG_BROKER_ID" ]] && [[ -n "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS" ]]; then
    old_IFS=$IFS
    IFS=','
    read -r -a voters <<< "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
    IFS=${old_IFS}
    broker_id_matched=false
    for voter in "${voters[@]}"; do
        if [[ "$voter" == *"$KAFKA_CFG_BROKER_ID"* ]]; then
            broker_id_matched=true
            break
        fi
    done

    if [[ "$broker_id_matched" == false ]]; then
        warn "KAFKA_CFG_BROKER_ID must match what is set in KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
    fi
fi
```

**Review by rafariossaa — APPROVED** [review]
LGTM

File: bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh
Comment (OneCricketeer) [inline bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:285]: This is not true; `process.roles=broker` is a valid config

Comment (carrodher) [reply bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:285]: Thanks for reporting this issue. Would you like to contribute by creating a PR to solve the issue? The Bitnami team will be happy to review it and provide feedback. [Here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md) you can find the contributing guidelines.

**Pull request merged**
